### PR TITLE
Add `use-true-false` rule (fixes #77)

### LINKED ISSUE #77: Detect boolean assertions, mandate `t.true()`
Assertions like this evaluate to `true` or `false`:

```js
t.truthy(array.indexOf(value) !== -1);
```

We should require you to write them like:

```js
t.true(array.indexOf(value) !== -1);
```

### PR DESCRIPTION
**Pull request by jfmengels**
Adds `use-true` rule (fixes #77)

### CODE REVIEW

**Comment by sindresorhus** [comment]
The name is a bit misleading as it supports `t.false` too.

Some other suggestions:

- `use-true-false`
- `use-strict-equal`
- `use-strict-equality`

**Comment by jfmengels** [comment]
Yeah, I changed it to `use-true-false`

File: docs/rules/use-true-false.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Ensure that `t.true()`/`t.false()` are used instead of `t.truthy()`/`t.falsy()`
+
+`t.true()` and `t.false()` are stricter in their checks than `t.truthy()` and `t.falsy`. This rule enforces the use of the former when the tested expression is known to result in a boolean value.
```
Comment (forresst) [inline docs/rules/use-true-false.md]: For a better understanding, I would add the example of @sindresorhus:

```markdown
`t.true()` and `t.false()` are stricter in their checks than `t.truthy()` and `t.falsy`.
For example: imagine you have a function foo() which normally returns true, but suddenly returns 1 instead. t.truthy(foo()) would not catch that case.
This rule enforces the use of the former when the tested expression is known to result in a boolean value.
```

File: docs/rules/use-true-false.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# Ensure that `t.true()`/`t.false()` are used instead of `t.truthy()`/`t.falsy()`
+
+`t.true()` and `t.false()` are stricter in their checks than `t.truthy()` and `t.falsy`.
+For example: if you have a function `foo()` which normally returns true, but suddenly returns 1 instead, `t.truthy(foo())` would not catch the change, but `t.true(foo())` would.
```
Comment (novemberborn) [inline docs/rules/use-true-false.md]: Maybe use backticks around "true" and "1"?

**Comment by novemberborn** [comment]
Very cool @jfmengels!

**Comment by sindresorhus** [comment]
Maybe also support builtin methods known to return booleans? I'm thinking `Array.isArray()`, etc.

**Comment by jfmengels** [comment]
@sindresorhus Would be nice yeah. Do you have a package that lists them by any chance?

**Comment by sindresorhus** [comment]
No

**Comment by sindresorhus** [comment]
Landed this. We can add improvements later: https://github.com/sindresorhus/eslint-plugin-ava/issues/79

Thanks @jfmengels! You're amazingly productive at this.

leo-clapping

**Comment by jfmengels** [comment]
You're welcome ;)

**Pull request merged**